docs/log_contexts.md

```diff
@@ -419,16 +404,15 @@ def do_some_stuff():
     return pcd
 ```
 
-It turns out that, thanks to the way that Deferreds chain together, we
+It turns out that, thanks to the way that awaitables chain together, we
 automatically get the property of a context-preserving deferred with
-`defer.inlineCallbacks`, provided the final Defered the function
-`yields` on has that property. So we can just write:
+`async`, provided the final Defered the function `yields` on has that
+property. So we can just write:
 
 ```python
-@defer.inlineCallbacks
-def handle_request():
-    yield do_some_stuff()
-    yield do_some_more_stuff()
+async def handle_request():
+    await do_some_stuff()
+    await do_some_more_stuff()
 ```
 
 To conclude: I think this scheme would have worked equally well, with
```

Comment:
can you `s/yield/await` and `s/inlineCallbacks/async` in the "garbage-collection" section below? afaik it otherwise all still applies.

Reply:
Are you sure? It says it's because "`await_something_interesting` is a generator function" which it doesn't seem to be? Or is an awaitable just a fancy generator?

Reply:
A coroutine is indeed, more-or-less, a fancy generator. Indeed, until Python 3.5, that's exactly what it was. The `async` syntax in Python 3.5 means that they aren't quite the same thing, but the mechanism is still basically the same.

After all, they amount to the same thing: a function gets halfway through, and is then suspended.